pyuvdata/uvdata/src/miriad_wrap.pyx

```diff
@@ -0,0 +1,578 @@
+  void hreadl_c(int item, long *buf, int offset, int length, int *iostat)
+
+  void hwriter_c(int item, float *buf, int offset, int length, int *iostat)
+  void hreadr_c(int item, float *buf, int offset,int length, int *iostat)
+
+  void hreadd_c(int item, double *buf, int offset, int length, int *iostat)
+  void hwrited_c(int item, double *buf, int offset, int length, int *iostat)
+
+  void hwritec_c(int item, float buf[2], int offset, int length, int *iostat)
+  void hreadc_c(int item, float buf[2], int offset, int length, int *iostat)
+
+# 4 is the ITEM_HDR_SIZE in both miriad3 and 4 but cython does
+# not like using it as the length of the array size it needs to be static
+# and not defined at compile time.
+cdef extern from "io.h":
+  cdef int ITEM_HDR_SIZE "ITEM_HDR_SIZE"
+  cdef char binary_item[4]
+  cdef char real_item[4]
+  cdef char int_item[4]
+  cdef char int2_item[4]
+  cdef char int8_item[4]
+  cdef char char_item[4]
+  cdef char dble_item[4]
+  cdef char cmplx_item[4]
+
+cdef extern from "hio.h":
+  int mroundup(int a, int b)
+
+cdef extern from "maxdimc.h":
+  cpdef int _MAXCHAN "MAXCHAN"
+
+MAXCHAN = _MAXCHAN
+
+cdef numpy.int_t MAXVAR = 32768
+ctypedef numpy.int_t DTYPE_t
+ctypedef numpy.complex64_t DTYPE_c
+ctypedef numpy.float64_t DTYPE_f64
+ctypedef numpy.int64_t DTYPE_sl
+
+cdef inline int GETI(int bl):
+  return (bl - 65536) / 2048 - 1 if bl > 65536 else (bl >> 8) - 1
+
+cdef inline int GETJ(int bl):
+  return (bl - 65536) % 2048 -1 if bl > 65536 else (bl & 255) - 1
+
+cdef inline float MKBL(int i, int j):
+  return (i + 1) << 8 | (j + 1) if (i + 1 < 256 and j + 1 < 256) else (i + 1) * 2048 + (j + 1 + 65536)
+
+cdef inline void CHK_IO(int i) except *:
+  if (i != 0):
+    raise IOError("IO failed.")
+
+class MiriadError(RuntimeError):
+  pass
+
+cdef int raise_miriad_error() except *:
+  raise MiriadError()
+
+cpdef hdaccess(int item_hdl) except +raise_miriad_error:
+  cdef int iostat
+  hdaccess_c(item_hdl, &iostat)
+  return
+
+cdef INIT(item_hdl, char type_item[4], size):
+  cdef int offset, iostat
+  hwriteb_c(item_hdl, type_item, 0, ITEM_HDR_SIZE, &iostat)
+  CHK_IO(iostat)
+  offset = mroundup(ITEM_HDR_SIZE, size)
+  return offset
+
+cpdef hwrite_init(int item_hdl, str type) except +raise_miriad_error:
+  cdef int offset
+
+  if type[0] == "a":
+    offset = INIT(item_hdl, char_item, H_BYTE_SIZE)
+
+  elif type[0] == "b":
+    offset = INIT(item_hdl, binary_item, ITEM_HDR_SIZE)
+
+  elif type[0] == "i":
+    offset = INIT(item_hdl, int_item, H_INT_SIZE)
+
+  elif type[0] == "j":
+    offset = INIT(item_hdl, int2_item, H_INT2_SIZE)
+
+  elif type[0] == "l":
+    offset = INIT(item_hdl, int8_item, H_INT8_SIZE)
+
+  elif type[0] == "r":
+    offset = INIT(item_hdl, real_item, H_REAL_SIZE)
+
+  elif type[0] == "d":
+    offset = INIT(item_hdl, dble_item, H_DBLE_SIZE)
+
+  elif type[0] == "c":
+    offset = INIT(item_hdl, cmplx_item, H_CMPLX_SIZE)
+
```

Comment:
Should we have an `else` block here for any other character? Or is that type of handling done somewhere else?

Reply:
I'll add that in, a lot of other functions have it. this one should too.